### PR TITLE
feat: `msgsSlice` store slice

### LINKED ISSUE #1159: [EIP-1271] Fetching of messages
## What is the feature about

Implementation of EIP-1271 "Messages" store/polling.

## The list of requirements

- Create "messages" slice
- Create `useInitMessages` poller hook
- Update the stubbed `useMessages` hook that returns mocked data from the accordion PR to use the store

## Links

- [Problem statement/data structure](https://www.notion.so/gnosis-safe/EIP-1271-signatures-can-only-be-generated-on-chain-2800b5e516124d959c38ba8f2042f9f7?d=e8ca1d332df54ae587e66240fd40e230#a8489e3724234f75a9c3d389e323a725)

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Partially resolves #1159

## How this PR fixes it

A new `msgsSlice` store slice has been created and associated hooks created/updated: `useLoadMessages`, `useMessages`.

## How to test it

Open the "Messages" in the transaction list and observe that two mock messges are there.

### CODE REVIEW

**Review by usame-algan — APPROVED** [review]
Nice work 👍

File: src/store/msgsSlice.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    owner: AddressEx
+    signature: string
+  }[]
+  proposedBy: AddressEx
+  preparedSignature: string | null // Name subject to change
+}
+
+export type MessageListPage = Page<MessageDateLabel | Message>
+
+const { slice, selector } = makeLoadableSlice('msgs', undefined as MessageListPage | undefined)
```
Comment (katspaugh) [inline src/store/msgsSlice.ts:48]: > msgs

Please rename it to something more readable.

Comment (iamacook) [reply src/store/msgsSlice.ts:48]: Every instance of `msg`/`msgs` variable(s) or just the slice? We shorten transactions to `tx`, hence my reasoning.

Comment (katspaugh) [reply src/store/msgsSlice.ts:48]: Just the slice name, because that's what you see in the Redux debugger.

Comment (katspaugh) [reply src/store/msgsSlice.ts:48]: The word messages in general is too broad for this feature, in my opinion. It's unfortunate that it's now spread across the code.

Comment (iamacook) [reply src/store/msgsSlice.ts:48]: I've adjusted it in the open EIP-1271-related PR [here](https://github.com/safe-global/web-core/pull/1224/commits/09db7c73e7ccff9d526723d37ce7b44e7be9858a).

Comment (iamacook) [reply src/store/msgsSlice.ts:48]: > The word messages in general is too broad for this feature, in my opinion. It's unfortunate that it's now spread across the code.

It'ss renamable but it is/was the agreed upon term. Do you have other suggestions?

**Pull request merged**